packages/catalog-model/src/kinds/relations.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+Naming rules for relations in priority order:
+
+1. Use at most two words. One main verb and a specifier, e.g. "ownerOf"
+2. Reading out "<source-kind> <type> <target-kind>" should make sense in English.
+3. Maintain symmetry between pairs, e.g. "ownedBy" and "ownerOf" rather than "owns".
+*/
+
+/**
+ * An ownership relation where the owner is usually an organizational
+ * entity (user or group), and the other entity can be anything.
+ */
+export const RELATION_OWNED_BY = 'ownedBy';
+export const RELATION_OWNER_OF = 'ownerOf';
+
+/**
+ * A relation with an API entity, typically from a component or system
+ */
+export const RELATION_IMPLEMENTED_BY = 'implementedBy';
+export const RELATION_IMPLEMENTS = 'implements';
```

Comment:
btw, we had some discussions on our side and think that "implemented" is probably not the perfect term in all cases. Implements could mean that one writes a server, but could also mean that someone writes a client.
We think providedBy/provides and the matching consumedBy/consumes would be a better long term naming.

(but this is not really related to this and I think that the new code should make a migration easier)

Reply:
Yeah we also discussed this one a bit. Another option brought up was `exposedBy`/`exposes`

Reply:
The words expose and consume sound very good together.

Reply:
Possibly `exposes` and `consumes` then?

I'm now thinking the original `provides`/`consumes` is my favorite. That wording does show up in software engineering in general when describing similar types of relationships.

Reply:
That's a fine choice.

I wonder what these will be for resources. Uses as a general term, or several more precise ones? component.uses.resource[type=database], resource[type=database].usedBy.component